server.js

```diff
@@ -257,6 +263,8 @@ app.use('/pl/course_instance/:course_instance_id/instructor', require('./middlew
 app.use('/pl/course_instance/:course_instance_id/instructor', function(req, res, next) {res.locals.urlPrefix = '/pl/course_instance/' + req.params.course_instance_id + '/instructor'; next();});
 app.use('/pl/course_instance/:course_instance_id/instructor', function(req, res, next) {res.locals.navbarType = 'instructor'; next();});
 app.use('/pl/course_instance/:course_instance_id/instructor', require('./middlewares/selectOpenIssueCount'));
+app.use('/pl/course_instance/:course_instance_id/instructor', require('./middlewares/selectCourseInstances'));
+app.use('/pl/course_instance/:course_instance_id/instructor', require('./middlewares/selectCourses'));
```

Comment:
We've generally been trying to not have too many middlewares that run DB queries, to avoid the overhead of making lots of small queries on every page load. The alternative is to add more things to the queries in `authzCourse/Instance` and `selectAndAuthz*`. Of course the disadvantage is that those middlewares get pretty complex, but at least they are fast 😄

I'm not necessarily saying that we should integrate `selectCourses` and `selectCourseInstances` into `authzCourse/Instance`, but it's worth considering.

Reply:
Done